vm/rust/src/jsonrpc.rs

```diff
@@ -298,32 +322,34 @@ impl From<OrderedL2ToL1Message> for OrderedMessage {
 pub struct Retdata(pub Vec<StarkFelt>);
 
 fn make_state_diff(
-    state: &mut TransactionalState<JunoStateReader>,
-    deprecated_declared_class: Option<ClassHash>,
+    state: &mut TransactionalState<CachedState<JunoStateReader>>,
+    deprecated_declared_class_hash: Option<ClassHash>,
 ) -> Result<StateDiff, StateError> {
-    let diff = state.to_state_diff();
+    let diff: CommitmentStateDiff = state.to_state_diff()?.into();
     let mut deployed_contracts = Vec::new();
     let mut replaced_classes = Vec::new();
 
-    for pair in diff.address_to_class_hash {
-        let existing_class_hash = state.state.get_class_hash_at(pair.0)?;
+    for (addr, class_hash) in diff.address_to_class_hash {
+        let existing_class_hash = state.state.get_class_hash_at(addr)?;
+        let addr: StarkFelt = addr.into();
+
         if existing_class_hash == ClassHash::default() {
             #[rustfmt::skip]
             deployed_contracts.push(DeployedContract {
-                address: *pair.0.0.key(),
-                class_hash: pair.1.0,
+                address: addr,
+                class_hash: class_hash.0,
             });
         } else {
             #[rustfmt::skip]
             replaced_classes.push(ReplacedClass {
-                contract_address: *pair.0.0.key(),
-                class_hash: pair.1.0,
+                contract_address: addr,
+                class_hash: class_hash.0,
             });
         }
     }
 
     let mut deprecated_declared_classes = Vec::default();
-    if let Some(v) = deprecated_declared_class {
+    if let Some(v) = deprecated_declared_class_hash {
```

Comment:
Sorry, another nit: deprecated_declared_classes -> deprecated_declared_class_hashes

Reply:
otherwise this file LGTM

Reply:
No worries. Fixed